GLTFSerialization/Tests/GLTFSerializationTests/GLBBuilderTest.cs

```diff
@@ -0,0 +1,343 @@
+			uint initialGLBLength = glbObject.BinaryChunkInfo.Length;
+
+			MemoryStream stream = new MemoryStream();
+			StreamWriter writer = new StreamWriter(stream);
+			writer.Write(TestAssetPaths.MIN_GLTF_STR);
+			writer.Flush();
+			stream.Position = 0;
+
+			GLTFRoot newRoot;
+			GLTFParser.ParseJson(stream, out newRoot);
+			GLBBuilder.SetRoot(glbObject, newRoot);
+			GLBBuilder.UpdateStream(glbObject);
+			Assert.AreEqual(glbObject.Header.FileLength, glbObject.JsonChunkInfo.StartPosition + glbObject.JsonChunkInfo.Length + GLTFParser.CHUNK_HEADER_SIZE);
+			glbOutStream.Close();
+			glbOutStream = new FileStream(outPath, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite);
+			List<ChunkInfo> chunks = GLTFParser.FindChunks(glbOutStream);
+			Assert.AreEqual(1, chunks.Count);
+
+			glbOutStream.Position = 0;
+			GLBObject testObject = GLBBuilder.ConstructFromStream(glbOutStream);
+
+			Assert.AreEqual(glbObject.JsonChunkInfo.Length, testObject.JsonChunkInfo.Length);
+			Assert.AreEqual(glbObject.BinaryChunkInfo.Length, testObject.BinaryChunkInfo.Length);
+		}
+
+		private BufferViewId AddBinaryDataToStreamHelper(GLBObject glbObject, Stream blobToAdd)
+		{
+			int previousCount = 0;
+			if (glbObject.Root.BufferViews != null)
+			{
+				previousCount = glbObject.Root.BufferViews.Count;
+			}
+
+			uint previousGLBLength = glbObject.Header.FileLength;
+			uint previousChunkLength = glbObject.BinaryChunkInfo.Length;
+
+			uint bufferSize = GLBBuilder.CalculateAlignment((uint)blobToAdd.Length, 4);
+			BufferViewId bufferViewId = GLBBuilder.AddBinaryData(glbObject, blobToAdd);
+
+			uint headerModifier = previousChunkLength == 0 ? GLTFParser.CHUNK_HEADER_SIZE : 0;
+			Assert.AreEqual(previousCount + 1, glbObject.Root.BufferViews.Count);
+			Assert.AreEqual(previousCount, bufferViewId.Id);
+			Assert.AreEqual(previousGLBLength + bufferSize + headerModifier, glbObject.Header.FileLength);
+			Assert.AreEqual(previousChunkLength + bufferSize, glbObject.BinaryChunkInfo.Length);
+			Assert.AreEqual(previousChunkLength + bufferSize, glbObject.Root.Buffers[0].ByteLength);
+			Assert.AreEqual(glbObject.Header.FileLength, glbObject.Stream.Length);
+
+			return bufferViewId;
+		}
+
+		private void CompareBinaryData(GLBObject resultObject, FileStream sourceStream)
+		{
+			MemoryStream outStream = new MemoryStream();
+			GLBObject sourceGLB = GLBBuilder.ConstructFromStream(sourceStream, outStream);
+			byte[] resultObjectBinary = new byte[resultObject.BinaryChunkInfo.Length];
+			resultObject.Stream.Position = resultObject.BinaryChunkInfo.StartPosition;
+			resultObject.Stream.Read(resultObjectBinary, 0, resultObjectBinary.Length);
+
+			byte[] sourceObjectBinary = new byte[sourceGLB.BinaryChunkInfo.Length];
+			sourceGLB.Stream.Position = sourceGLB.BinaryChunkInfo.StartPosition;
+			sourceGLB.Stream.Read(sourceObjectBinary, 0, sourceObjectBinary.Length);
+
+			Assert.IsTrue(resultObjectBinary.SequenceEqual(sourceObjectBinary));
+		}
+	}
```

Comment:
Would it be worth adding a slightly broader test to serialize and deserialize, and compare the hierarchy after a round-trip?

Reply:
I'll update CreateAndSaveFromEmptyStream to do a JSON check